src/khepri_fun.erl

```diff
@@ -283,45 +285,59 @@ handle_compilation_error(
   Asm,
   {error,
    [{_GeneratedModuleName,
-     [{_, beam_validator,
-       {FailingFun,
-        {{get_tuple_element, Src, _Element, _Dst},
-         _,
-         {bad_type,
-          {needed, {t_tuple, _Size, _, _Fields} = NeededType},
-          {actual, any}}}}}]}],
-   []} = Error) ->
-    VarInfo = {var_info, Src, [{type, NeededType}]},
-    Comment = {'%', VarInfo},
-    add_comment_and_retry(Asm, Error, FailingFun, Comment);
+     [{_, beam_validator, ValidationFailure} | _Rest]}],
+  []} = Error) ->
+    handle_validation_error(Asm, ValidationFailure, Error);
 handle_compilation_error(
   Asm,
   %% Same as above, but returned by Erlang 23's compiler instead of Erlang 24+.
   {error,
    [{_GeneratedModuleName,
-     [{beam_validator,
-       {FailingFun,
-        {{get_tuple_element, Src, _Element, _Dst},
-         _,
-         {bad_type,
-          {needed, {t_tuple, _Size, _, _Fields} = NeededType},
-          {actual, any}}}}}]}],
-   []} = Error) ->
-    VarInfo = {var_info, Src, [{type, NeededType}]},
-    Comment = {'%', VarInfo},
-    add_comment_and_retry(Asm, Error, FailingFun, Comment);
+     [{beam_validator, ValidationFailure} | _Rest]}],
+  []} = Error) ->
+    handle_validation_error(Asm, ValidationFailure, Error);
 handle_compilation_error(Asm, Error) ->
     throw({compilation_failure, Error, Asm}).
 
+handle_validation_error(
+  Asm,
+  {_FailingFun,
+   {{Call, 1 = _Arity, {f, EntryLabel}},
+    _Index,
+    no_bs_start_match2}},
+  _Error) when Call =:= call orelse Call =:= call_only ->
+    %% TODO: hard-coded register
+    Comment = {'%', {var_info, {x, 0}, [accepts_match_context]}},
+    %% The first three instructions are always
+    %% - an initial label
+    %% - the func_info instruction
+    %% - the function's entry-point label
```

Comment:
Hmm good point, I think we can add annotations here by inserting before the `FailingInstruction` in `{FailingFun, {FailingInstruction, Location, Failure}}` instead of using `Location`.